br/pkg/streamhelper/advancer_test.go

```diff
@@ -548,8 +626,10 @@ func TestCheckPointLagged(t *testing.T) {
 	})
 	adv.StartTaskListener(ctx)
 	c.advanceClusterTimeBy(2 * time.Minute)
+	// if global ts is not advanced, the checkpoint will not be lagged
```

Comment:
```
	if globalTs <= c.task.StartTs {
		// task is not started yet
		return false, nil
	}
```

Then maybe here should be < instead of <= ?

Reply:
if `globalTs == c.task.StartTs` this will only happen when new task created. 

Since it's not the common case after task running for some time. I think it's better to make this not pause by default

Reply:
Sometimes the task may be stuck from creating, say, the advancer doesn't work or one of TiKV didn't notice the task.

Reply:
After talk with @YuJuncen , we had a agreement to make the check include when `globalTs == task.StartTs`. I changed it.


Additionally I found the unproper error return logic when add task. I also fixed it this PR, and fix the related test cases.